src/Datadog.Trace.ClrProfiler.Managed/Integrations/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -121,8 +146,39 @@ public static class ConnectionMultiplexer
             int mdToken,
             long moduleVersionPtr)
         {
-            var callOpCode = (OpCodeValue)opCode;
-            return ExecuteAsyncImplInternal<T>(multiplexer, message, processor, state, server, callOpCode);
+            if (multiplexer == null)
+            {
+                throw new ArgumentNullException(nameof(multiplexer));
+            }
+
+            var genericType = typeof(T);
+            var multiplexerType = multiplexer.GetInstrumentedType(ConnectionMultiplexerTypeName);
+            Func<object, object, object, object, object, Task<T>> instrumentedMethod;
+
+            try
+            {
+                instrumentedMethod =
```

Comment:
Not gonna block the PR on this, but I think the `MethodBuilder` call should have stayed in `ExecuteAsyncImplInternal()`. The idea with the "double wrapper" implementation was that the method with `[InterceptMethod]` would only be a very lightweight wrapper (2-3 lines of code) around the real implementation. It exists only because we can't use `Task` or `Task<T>` in the method signature which is required to use `async`/`await` method. Instead, the "inner" implementation can use `async`/`await` to set up the continuation and return `Task<T>`.

Reply:
If we end up going with the Continue with API, could we ultimately cast and return it as an object?